estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -182,7 +182,7 @@ class AssociatedDocument(Generic[_BaseDocument]):
     doc: _BaseDocument
     binding: int
 
-FetchSnapshotFn = Callable[[Logger], AsyncGenerator[_BaseDocument | dict, None]]
+FetchSnapshotFn = Callable[[Logger, Task], AsyncGenerator[_BaseDocument | dict, None]]
```

Comment:
You don't need to add Task here (and, if you're doing so, this PR needs to fix all other connectors which were broken).

Instead, just close over Task in the code path that's calling open_binding()

Reply:
That's what I thought too, and I was confused when it didn't seem to be emitting any documents: it's the wrong Task. The Task [passed to `resource.open`](https://github.com/estuary/connectors/blob/3888fc81a225bf8d38d8780cdd06713f511dffd0/estuary-cdk/estuary_cdk/capture/common.py#L399-L405) is actually the _parent_ of the Task that actually runs the generators. It's actually `open_binding` itself that [creates](https://github.com/estuary/connectors/blob/main/estuary-cdk/estuary_cdk/capture/common.py#L438) [the](https://github.com/estuary/connectors/blob/main/estuary-cdk/estuary_cdk/capture/common.py#L448) [final](https://github.com/estuary/connectors/blob/main/estuary-cdk/estuary_cdk/capture/common.py#L463) Tasks that contain the buffers we want to write to here (and it has to, as we need separate a separate Task/buffer per generator type -- backfill, incremental, snapshot)

Reply:
:man_facepalming: d'oh, you're totally right.

Okay. Trying to figure out how to not make this more confusing for the common case.

One option would be to replace the Logger parameter with Task, since the Logger is trivially accessed as `task.log` (and this PR does then need to update all call sites). That's ... okay, but if I'm writing a connector, I'm wondering what this Task is and what I'm supposed to do with it ("do I call `captured` myself?").

Or, could the netsuite connector just re-implement open_binding itself? If you did that and called _binding_backfill_task / _binding_incremental_task directly it could work?

Reply:
>Or, could the netsuite connector just re-implement open_binding itself?

Yep, good point. Removed this commit in favor of re-implementing `open_binding` in the connector